src/query/functions/src/aggregates/aggregate_string_agg.rs

```diff
@@ -44,6 +49,51 @@ pub struct StringAggState {
     values: String,
 }
 
+#[derive(BorshSerialize, BorshDeserialize, Debug, Default)]
+pub struct SortStringAggState {
+    inner: ArrayAggState<StringType>,
+    delimiter: String,
+    sort_desc: Option<AggregateFunctionSortDesc>,
+}
+
+impl ScalarStateFunc<StringType> for SortStringAggState {
+    fn new() -> Self {
+        Self::default()
+    }
+
+    fn add(&mut self, other: Option<&'_ str>) {
+        self.inner.add(other);
+    }
+
+    fn add_batch(&mut self, column: &StringColumn, validity: Option<&Bitmap>) -> Result<()> {
+        self.inner.add_batch(column, validity)
+    }
+
+    fn merge(&mut self, rhs: &Self) -> Result<()> {
+        self.inner.merge(&rhs.inner)
+    }
+
+    fn merge_result(&mut self, builder: &mut ColumnBuilder) -> Result<()> {
+        if let Some(opts) = self.sort_desc {
+            self.inner.values.sort_by(|a, b| {
```

Comment:
better to build a `DataBlock` and use `sort` function to do the order by.
https://github.com/databendlabs/databend/blob/main/src/query/expression/src/kernels/sort.rs#L77